### DISCUSSION TITLE
How to use search-field in Invoice/Offer?

### DISCUSSION THREAD

**Discussion by mad99** [YetiForceCompany/YetiForceCRM#14374] [How to] [answered]
Hi,
Would be great to get a hint, push in the right direction.
I would like to use the search-/product-suggestion. But this was not working since I use YetiForce. 

- Maybe I did something wrong in the products? 
- Maybe I misunderstood some basics?

I would like to type some characters (the beginning of some ore multi products in the YetForce-product-Database) in the box with the description «Suchbegriff eintippen» =>en: "type query". I guess I should get some like a suggestion-list back: From products that have the typed characters in the Product-Name.

### Problem
But no matter what I type, it always comes up«Keine Ergebnisse gefudnen» => en: "No results found".

![YetiForce_search-field](https://user-images.githubusercontent.com/6169308/110463526-51df2c80-80d2-11eb-839f-4f4d32cda959.png)

I've gotten into the habit of clicking the magnifying glass so much that I almost click it automatically without typing anything. But with the upgrade to 6.1, I thought maybe that would change!

But it is still the same unexpected behaviour as in version 5.
Maybe I set something wrong in the settings...? or I have to pay attention to something in the products?

### Question

- Does anybody know the reason/had the same problem and found a solution?
- Or did i misunderstood the Expected actions or needed steps to find products?
- Or does anybody has an idea what I have to change in YetiForce or the general-settings or in the Products, so the field will get a list of products with matching product-name in it?

**Comment by alexgrndprx** [comment, marked as answer]
It works on https://gitdeveloper.yetiforce.com 
Check this setting https://gitdeveloper.yetiforce.com/index.php?module=Search&parent=Settings&view=Index 
And check CRON job "Update labels" https://gitdeveloper.yetiforce.com/index.php?module=CronTasks&parent=Settings&view=List 
and this table "u_yf_crmentity_search_label" it should be filled with product names (or with any other fields you want to search by)

**Reply by mad99** [reply]
Thank you @alexgrndprx
Looks great on gitdevelper.yetiforce.com

1. I have checked the Settings (Seams to be similar) but is there some important part that I have to look for?
![YetiForce_Cursor_und_Suchkonfiguration_System_Einstellungen](https://user-images.githubusercontent.com/6169308/110758140-73642380-824c-11eb-812e-b213bf4125f2.png)


2. I have checked CRON 

> _(Unfortunately when I activate all of them => I run in the old bug-report https://github.com/YetiForceCompany/YetiForceCRM/discussions/14156#discussioncomment-340387 with Missing Inventory-part. This time I have also tried the hint from @vovpff and   @Giorgos-Hellene. But the right part is deleted.
So, I played back the backup-Files for File and DB, and the invoice-right-part was back. Then I deactivate all cron-jobs and restart the cron. The right-part-of-invoice is still there. So I'm quite sure, one point of the cronjobs is crashing my system. So I only activated the cronjobs which I suggested, they have some connection to the need products and the search for them)_

![YetiForce_Cursor_und_CRON_Aufgaben_System_Einstellungen](https://user-images.githubusercontent.com/6169308/110758897-52e89900-824d-11eb-9327-98909008e6ee.png)


They are running without breaking my system and delete the right part of the invoice. But in the output it seams something is not smooth:

<img width="876" alt="YetiForce_CronJobsOutput" src="https://user-images.githubusercontent.com/6169308/110759235-ad81f500-824d-11eb-93d4-a290eebc6857.png">


### But 
3. I also checked the DB: In the Database, the table «u_yf_crmentity_search_label» is filled also with product-names that I would like to find in the Product-Field-Search: 

![YetiForce_MySQL_5_7_30-log__Infomaniak_SSH_Tunnel_zu_MySql_ku1ls_ye_six_6221_smarter_design_u_yf_crmentity_search_label](https://user-images.githubusercontent.com/6169308/110759843-57fa1800-824e-11eb-96d1-1f87be4730f7.png)

## Questions
Maybe there is some other CronJob that I have to activate, so the search-field is connected and will get some matches?

- Does anybody has a hint about the missing cronjob? _(Do I have to do something different so the output looks with all of points on «Start task» and «End task?)_
- Or is there an important Setting in Search-Module that I have to change or look especially on?
- Or does anybody know other points/helps that could made it possible for YetiForce to have some match in the search-field of the products?


![YetiForce_Search-Filed-Products](https://user-images.githubusercontent.com/6169308/110760237-ca6af800-824e-11eb-89fe-37ad45e21dc5.png)

**Reply by alexgrndprx** [reply]
Check request/response after typing 3+ char in search field, they should look like this

![rq1](https://user-images.githubusercontent.com/79980057/110765222-a44c5480-825c-11eb-8291-6e4a8c1c52c5.PNG)
![res1](https://user-images.githubusercontent.com/79980057/110765213-a31b2780-825c-11eb-847d-4242fd32b556.PNG)

also, have you tried searching in other modules?
![search](https://user-images.githubusercontent.com/79980057/110765672-1755cb00-825d-11eb-8326-41827151935e.PNG)

**Reply by rskrzypczak** [reply]
Two cron jobs are required:
![image](https://user-images.githubusercontent.com/10221999/110767849-f7bba400-8256-11eb-8a98-eb4c950ba121.png)

**Reply by mad99** [reply]
GREEEEAAAAT!!!!
@alexgrndprx & @rskrzypczak you are my heros of the day/week!!!

I checked the Request & Response. There was a there was the _csrf in front of the query. 
Thought maybe some problem with session-id?

![Bildschirmfoto 2021-03-11 um 11 05 27](https://user-images.githubusercontent.com/6169308/110771486-d6f54d80-825a-11eb-941b-fbaa3d9e84e3.png)


and the result was [] empty.

![Bildschirmfoto 2021-03-11 um 11 05 49](https://user-images.githubusercontent.com/6169308/110771574-effdfe80-825a-11eb-93a2-08517e955599.png)


### Than @rskrzypczak's hint with the «**Update permissions**» cron jobs.
## And it rocks!!

![Bildschirmfoto 2021-03-11 um 11 13 13](https://user-images.githubusercontent.com/6169308/110771588-f55b4900-825a-11eb-9ebe-52890e56ab13.png)


# The Best:
**The system is not broken. This time the right-part of the invoice is still there. So I can do the cron without problems and now I have the List-Suggestion.**

Thanks a lot for help!
Martin

**Comment by mad99** [comment]
Hey @alexgrndprx & @rskrzypczak 
you are my heros of the day/week.

First I checked the Request & Responses.
![Bildschirmfoto 2021-03-11 um 11 05 27](https://user-images.githubusercontent.com/6169308/110774139-c4304800-825d-11eb-8f7b-ee196fee5b02.png)

There was only the _csrf part in front and I suggested, maybe it have something to do with session-ID and rights.
Response is empty []

![Bildschirmfoto 2021-03-11 um 11 05 49](https://user-images.githubusercontent.com/6169308/110774296-f04bc900-825d-11eb-8211-aeac22766cbd.png)

Than the hint from @rskrzypczak about the permissions update cron job:

### and it ROCKS!!

![Bildschirmfoto 2021-03-11 um 11 13 13](https://user-images.githubusercontent.com/6169308/110774393-09547a00-825e-11eb-9128-dd912bda5315.png)

Now I have the Suggestion-List and I can choose from the list and use it in invoice, offers, ...

## And the best:
My system is still working. No crash after cronjob and loosing the right-part of the invoice! Great!!

So I have to aktivate at least the 4 cron jobs. So the Search-List is usable and the system still works.


![YetiForce_Cursor_und_CRON_Aufgaben_System_Einstellungen](https://user-images.githubusercontent.com/6169308/110774834-8384fe80-825e-11eb-9347-8a828dbb591d.png)

Thank you very much!!
Best regards
Martin
